app/src/main/java/petermcneil/musiclibrary/services/database/ArtistService.java

```diff
@@ -0,0 +1,161 @@
+    }
+
+    @Override
+    public List<Artist> getList() {
+        return jdbcTemplate.query("SELECT * FROM artist", new EmptySqlParameterSource(), new ResultSetExtractor<List<Artist>>() {
+            @Override
+            public List<Artist> extractData(ResultSet rs) throws SQLException, DataAccessException {
+                List<Artist> result = new ArrayList<>();
+                while(rs.next()){
+                    Artist.Builder artist =  Artist.artistBuilder();
+
+                    artist.artistId(rs.getInt("idartist"));
+                    artist.name(rs.getString("name"));
+                    artist.type(getType(rs.getInt("idartisttype")));
+                    artist.bio(bioService.get(rs.getInt("idbio")));
+
+                    result.add(artist.build());
+                }
+                return result;
+            }
+        });
+    }
+
+
+    @Override
+    public Integer post(Artist artist) {
+        MapSqlParameterSource params = new MapSqlParameterSource();
+
+        params.addValue("name", artist.getName());
+        params.addValue("idartisttype", getTypeId(artist.getType()));
+
+        LOG.info("REQUEST : Add a new bio for {} and return the key", artist.getName());
+        params.addValue("idbio", bioService.post(artist.getBio()));
+
+        KeyHolder key = new GeneratedKeyHolder();
+        jdbcTemplate.update("INSERT INTO artist(name, idartisttype, idbio) VALUES (:name, :idartisttype, :idbio)", params, key);
+
+        Map<String, Object> artistData = key.getKeyList().get(0);
+        Integer artistId = Integer.parseInt(artistData.get("idartist").toString());
+
+        LOG.info("RESPONSE: Passing back the id ({}) to the GET controller", artistId);
+        return artistId;
+    }
+
+    @Override
+    public void put(Artist artist, Integer artistId) {
+        MapSqlParameterSource params = new MapSqlParameterSource();
+
+        params.addValue("artistId", artistId);
+        params.addValue("artistName", artist.getName());
+        params.addValue("artistType", getTypeId(artist.getType()));
+
+        bioService.put(artist.getBio(), artist.getBio().getBioId());
+        params.addValue("artistBio", artist.getBio());
+
+        jdbcTemplate.update("UPDATE artist SET name=:artistName, idartisttype=:artistType WHERE idartist=:artistId", params);
+    }
+
+    @Override
+    public void delete(Integer artistId) {
+        MapSqlParameterSource bioParams = new MapSqlParameterSource();
+        bioParams.addValue("artistId", artistId);
+
+        Integer bioId = jdbcTemplate.queryForObject("SELECT idbio FROM artist WHERE idartist=:artistId", bioParams, Integer.class);
+
+
+        MapSqlParameterSource params = new MapSqlParameterSource();
+        params.addValue("artist", artistId);
+
+        jdbcTemplate.update("DELETE FROM artist WHERE idartist=:artist", params);
+
+        LOG.info("REQUEST : DELETE the bio at the id ({})", bioId);
+        bioService.delete(bioId);
+
```

Comment:
Hmm is bio and artist a 1 to 1 mapping? Might as well have bio as a field inside the artists table if so